compiler/rustc_trait_selection/src/traits/error_reporting/mod.rs

```diff
@@ -2000,28 +2018,17 @@ impl<'a, 'tcx> InferCtxtPrivExt<'a, 'tcx> for InferCtxt<'a, 'tcx> {
                         self.emit_inference_failure_err(
                             body_id,
                             span,
-                            subst,
+                            trait_ref.self_ty().skip_binder().into(),
                             vec![],
                             ErrorCode::E0282,
+                            false,
```

Comment:
how are the errors impacted by using `true` here?

Reply:
It points at the binding, which is not very helpful imo.

```
LL |     for i in Vec::new() { }
   |         -    ^^^^^^^^ cannot infer type of the type parameter `T` declared on the struct `Vec`
   |         |
   |         type must be known at this point
```